tests/unit_tests/test_matchmaker_queue.py

```diff
@@ -433,21 +452,37 @@ def make_matches(*args):
         time.sleep(0.2)
 
         is_matching = False
-        return []
+        return [], []
 
-    with mock.patch(
-        "server.matchmaker.matchmaker_queue.make_matches",
-        make_matches
-    ):
-        queues = [queue_factory(f"Queue{i}") for i in range(5)]
-        # Ensure that find_matches does not short circuit
-        for queue in queues:
-            queue._queue = {
-                mock.Mock(players=[1]): 1,
-                mock.Mock(players=[2]): 2
-            }
-            queue.find_teams = mock.Mock()
-
-        await asyncio.gather(*[
-            queue.find_matches() for queue in queues
-        ])
+    queues = [queue_factory(f"Queue{i}") for i in range(5)]
+    # Ensure that find_matches does not short circuit
+    for queue in queues:
+        queue._queue = {
+            mock.Mock(players=[1]): 1,
+            mock.Mock(players=[2]): 2
+        }
+        queue.find_teams = mock.Mock()
+        queue._register_unmatched_searches = mock.Mock()
+        queue.matchmaker.find = mock.Mock(side_effect=find)
+
+    await asyncio.gather(*[
+        queue.find_matches() for queue in queues
+    ])
+
+
+@pytest.mark.asyncio
+async def test_queue_pop_communicates_failed_attempts(matchmaker_queue, player_factory):
```

Comment:
As @BlackYps pointed out, this would currently fail if there was only 1 player in the queue.